pkg/report/table/misconfig.go

```diff
@@ -50,6 +50,11 @@ func NewMisconfigRenderer(result types.Result, severities []dbTypes.Severity, tr
 }
 
 func (r *misconfigRenderer) Render() string {
+	// Trivy doesn't currently support showing suppressed misconfigs
+	// So just skip this result
+	if len(r.result.Misconfigurations) == 0 {
+		return ""
+	}
```

Comment:
@simar7 @nikpivkin 
Do we need to hide table if any misconfigs have been ignored?

Reply:
The report displays misconfigurations as a list, not a table, with the header showing the number of ignored results.  I don't think this is worth hiding. @simar7 wdyt?

Reply:
right. It looks like that:
```
Dockerfile (dockerfile)

Tests: 27 (SUCCESSES: 26, FAILURES: 0, EXCEPTIONS: 1)
Failures: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```
but by default - if no `failures` are found - we hide it.

Reply:
@DmitriyLewen you mean this right?

```
$ trivy config .                                 
2024-07-16T22:12:14-06:00       INFO    Misconfiguration scanning is enabled
2024-07-16T22:12:15-06:00       INFO    Detected config files   num=0

$  empty-dir tree
.

0 directories, 0 files
```

If so, I think this behavior is fine as-is because it is inline with the other scanners:

```
 trivy fs .
2024-07-16T22:14:13-06:00       INFO    Vulnerability scanning is enabled
2024-07-16T22:14:13-06:00       INFO    Secret scanning is enabled
2024-07-16T22:14:13-06:00       INFO    If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2024-07-16T22:14:13-06:00       INFO    Please see also https://aquasecurity.github.io/trivy/v0.53/docs/scanner/secret#recommendation for faster secret detection
2024-07-16T22:14:13-06:00       INFO    Number of language-specific files       num=0

```

Reply:
@simar7 take a look the following example:

Dockerfile without misconfigs:
```bash
➜  cat Dockerfile           
FROM alpine:3.12.1

COPY 123.txt 123.txt

USER patrick

HEALTHCHECK --interval=5m --timeout=3s CMD curl -f http://localhost/ || exit 1 
➜  trivy config ./Dockerfile
2024-07-17T10:24:20+06:00	INFO	Misconfiguration scanning is enabled
2024-07-17T10:24:21+06:00	INFO	Detected config files	num=1
```
Dockerfile with misconfig:
```bash
➜  cat Dockerfile 
FROM alpine:3.12.1

ADD 123.txt 123.txt

USER patrick

HEALTHCHECK --interval=5m --timeout=3s CMD curl -f http://localhost/ || exit 1
 
➜  trivy config ./Dockerfile
2024-07-17T10:25:09+06:00	INFO	Misconfiguration scanning is enabled
2024-07-17T10:25:10+06:00	INFO	Detected config files	num=1

Dockerfile (dockerfile)

Tests: 27 (SUCCESSES: 26, FAILURES: 1, EXCEPTIONS: 0)
Failures: 1 (UNKNOWN: 0, LOW: 1, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```
Dockerfile with ignored misconfig  
```bash
➜  cat Dockerfile 
FROM alpine:3.12.1

ADD 123.txt 123.txt

USER patrick

HEALTHCHECK --interval=5m --timeout=3s CMD curl -f http://localhost/ || exit 1 

➜  cat .trivyignore.yaml 
misconfigurations:
  - id: AVD-DS-0005                                                           

➜ trivy config ./Dockerfile --ignorefile .trivyignore.yaml 
2024-07-17T10:26:56+06:00	INFO	Misconfiguration scanning is enabled
2024-07-17T10:26:56+06:00	INFO	Detected config files	num=1

Dockerfile (dockerfile)

Tests: 27 (SUCCESSES: 26, FAILURES: 0, EXCEPTIONS: 1)
Failures: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)

## Trivy after these changes
➜  ./trivy config ./Dockerfile --ignorefile .trivyignore.yaml
2024-07-17T10:42:43+06:00	INFO	Misconfiguration scanning is enabled
2024-07-17T10:42:44+06:00	INFO	Detected config files	num=1

```

Reply:
Right I understand the change, I'm just saying that we shouldn't hide the fact that we've ignored stuff. I guess the difference lies in the fact that other scanners don't have the `EXCEPTIONS` field while misconfigurations scanner does. 

IMO, showing more info to the user is useful and create less opportunity for confusion. In misconfiguration scanning, we show the user how many config files we've scanned via the `Detected config files num=1` log output. This was done to differentiate the case when there's no misconfiguration vs there was no relevant file to find misconfigurations in.

I'm not sure of the original motivation for this PR but when things are ignored and not shown, it can be due to either us not finding an issue or the issue getting ignored (by an ignore policy).

Reply:
> I'm not sure of the original motivation for this PR 

We hide empty tables to avoid very large output (as is the case with tables for language files without vulnerabilities).

>  I guess the difference lies in the fact that other scanners don't have the EXCEPTIONS field while misconfigurations scanner does.

right. That is why i asked you about this.

> we show the user how many config files we've scanned via the Detected config files num=1 log output.

We also show ignored files in `debug` log:
```bash
2024-07-17T11:19:03+06:00	DEBUG	Ignored	id="AVD-DS-0005" target="Dockerfile"
```

Just to make sure I understand you correctly - if all `failure` tests are ignored - we can hide the header with the number of tests (e.g. `Tests: 27 (SUCCESSES: 26, FAILURES: 0, EXCEPTIONS: 1)`), right?

Reply:
Hmm I see yeah I think based on what you said regarding showing debug log message and the fact it inlines the behavior with the other scanners, we should hide it.

But at the same time, maybe we should also drop support for `EXCEPTIONS` as a whole since no other scanner has a concept for it. Is it worth tracking ignored misconfigurations/vulnerabilities?

Reply:
> maybe we should also drop support for EXCEPTIONS as a whole since no other scanner has a concept for it.

Unfortunately, I can't correctly assess how necessary this is. That's why I turned to you :smile:

> Is it worth tracking ignored misconfigurations/vulnerabilities?

Users can check ignored vulns using `--show-suppressed` flag - https://aquasecurity.github.io/trivy/v0.53/docs/configuration/filtering/#suppression

Reply:
I personally think it's reasonable to drop support for Exceptions since:

1. No other scanner uses it today. So the concept of Exceptions only exists in misconfiguration scanning. I believe the original concept originated from [conftest](https://www.conftest.dev/exceptions/) but we have diverged quite a bit from that now in Trivy.
2. The fact that it isn't very useful once we start to hide misconfigurations, if they are ignored. 

It will obviously be a breaking change that we will need to announce.

What do you think @knqyf263 @itaysk?

Reply:
It sounds reasonable. It's more aligned with other scanners.

Reply:
We can then merge this PR and create a new issue to remove the Exceptions field.
@simar7 can you open new issue?

Reply:
Created an issue to track https://github.com/aquasecurity/trivy/issues/7314